### ISSUE TITLE
Updating from 1.3.2 to 1.4.25 results in errors creating SqlServer actors

### ISSUE THREAD

**Issue by BobbyMorris** [akkadotnet/Akka.Persistence.SqlServer#231]
**Version Information**
Version of Akka.NET? 1.4.27
Which Akka.NET Modules? Akka.Persistence.SqlServer 1.4.25

**Describe the bug**
After upgrading NuGets from 1.3.2 we wind up with exceptions being thrown from Microsoft.Data.SqlClient.

**To Reproduce**
1. Start with an Akka.NET 1.3.2 project with this HOCON:
```
akka {
  loglevel = DEBUG
  loggers = ["Akka.Logger.NLog.NLogLogger, Akka.Logger.NLog"]
  actor {
    debug {
      receive = on
      autoreceive = on
      lifecycle = on
      event-stream = on
      unhandled = on
    }
  }
}
      
akka.persistence {
  journal {
    plugin = "akka.persistence.journal.sql-server"
    sql-server {
      class = "Akka.Persistence.SqlServer.Journal.SqlServerJournal, Akka.Persistence.SqlServer"
      plugin-dispatcher = "akka.actor.default-dispatcher"

      # connection string used for database access
      connection-string = "Data Source=.\\SQLEXPRESS;Initial Catalog=MyApp;Integrated Security=True"

      # default SQL timeout
      connection-timeout = 30s

      # SQL server schema name
      schema-name = akka

      # persistent journal table name
      table-name = EventJournal

      # initialize journal table automatically
      auto-initialize = off

      timestamp-provider = "Akka.Persistence.Sql.Common.Journal.DefaultTimestampProvider, Akka.Persistence.Sql.Common"
      metadata-table-name = Metadata
    }
  }
        
  snapshot-store {
    plugin = "akka.persistence.snapshot-store.sql-server"
    sql-server {
      class = "Akka.Persistence.SqlServer.Snapshot.SqlServerSnapshotStore, Akka.Persistence.SqlServer"
      plugin-dispatcher = "akka.actor.default-dispatcher"
      table-name = SnapshotStore
      schema-name = akka
      auto-initialize = off
      connection-string = "Data Source=.\\SQLEXPRESS;Initial Catalog=MyApp;Integrated Security=True"
    }
  }
}
```
2. Then update Akka.NET NuGets to 1.4.27 / 1.4.25
3. Run project

**Expected behavior**
```
|DEBUG|Akka.Actor.SystemGuardianActor|now supervising akka://fuse/system/akka.persistence.journal.sql-server
|DEBUG|Akka.Actor.SystemGuardianActor|now supervising akka://fuse/system/akka.persistence.snapshot-store.sql-server
|DEBUG|Akka.Actor.GuardianActor|now supervising akka://fuse/user/$a
|DEBUG|Akka.Actor.SystemGuardianActor|now supervising akka://fuse/system/recoveryPermitter
|DEBUG|Akka.Persistence.RecoveryPermitter|Started (Akka.Persistence.RecoveryPermitter)
|DEBUG|Akka.Persistence.Journal.AsyncWriteJournal+Resequencer|Started (Akka.Persistence.Journal.AsyncWriteJournal+Resequencer)
|DEBUG|Akka.Persistence.SqlServer.Snapshot.SqlServerSnapshotStore|Started (Akka.Persistence.SqlServer.Snapshot.SqlServerSnapshotStore)
|DEBUG|Akka.Persistence.SqlServer.Snapshot.SqlServerSnapshotStore|Started (Akka.Persistence.SqlServer.Snapshot.SqlServerSnapshotStore)
|DEBUG|MyProj.MyActor|MyActor PreStart
|DEBUG|MyProj.MyActor|Started (MyProj.MyActor)
|DEBUG|MyProj.MyActor|now watched by [akka://fuse/system/recoveryPermitter#775293092]
|DEBUG|Akka.Persistence.SqlServer.Journal.SqlServerJournal|Started (Akka.Persistence.SqlServer.Journal.SqlServerJournal)
|DEBUG|Akka.Persistence.Journal.ReplayFilter|Started (Akka.Persistence.Journal.ReplayFilter)
|DEBUG|Akka.Persistence.SqlServer.Journal.SqlServerJournal|now supervising akka://fuse/system/akka.persistence.journal.sql-server/$a
|DEBUG|Akka.Persistence.Journal.ReplayFilter|Stopped
|DEBUG|MyProj.MyActor|no longer watched by [akka://fuse/system/recoveryPermitter#775293092]
```

**Actual behavior**
```
|DEBUG|Akka.Actor.SystemGuardianActor|now supervising akka://fuse/system/akka.persistence.journal.sql-server
|DEBUG|Akka.Actor.SystemGuardianActor|now supervising akka://fuse/system/akka.persistence.snapshot-store.sql-server
|DEBUG|Akka.Actor.SystemGuardianActor|now supervising akka://fuse/system/recoveryPermitter
|DEBUG|Akka.Persistence.RecoveryPermitter|Started (Akka.Persistence.RecoveryPermitter)
|DEBUG|MyProj.MyActor|Started (MyProj.MyActor)
|DEBUG|MyProj.MyActor|now watched by [akka://fuse/system/recoveryPermitter#1649681903]
|DEBUG|Akka.Actor.GuardianActor|now supervising akka://fuse/user/$a
|DEBUG|Akka.Persistence.Journal.AsyncWriteJournal+Resequencer|Started (Akka.Persistence.Journal.AsyncWriteJournal+Resequencer)
|ERROR|Akka.Actor.OneForOneStrategy|Error while creating actor instance of type Akka.Persistence.SqlServer.Snapshot.SqlServerSnapshotStore with 1 args: (  class : "Akka.Persistence.SqlServer.Snapshot.SqlServerSnapshotStore, Akka.Persistence.SqlServer"
  plugin-dispatcher : akka.actor.default-dispatcher
  connection-string : "Data Source=.\SQLEXPRESS;Initial Catalog=MyApp;Integrated Security=True"
  connection-timeout : 30s
  schema-name : akka
  table-name : SnapshotStore
  auto-initialize : off
  sequential-access : on
  use-constant-parameter-size : false
)
|ERROR|Akka.Actor.OneForOneStrategy|Error while creating actor instance of type Akka.Persistence.SqlServer.Journal.SqlServerJournal with 1 args: (  class : "Akka.Persistence.SqlServer.Journal.SqlServerJournal, Akka.Persistence.SqlServer"
  plugin-dispatcher : akka.actor.default-dispatcher
  connection-string : "Data Source=.\SQLEXPRESS;Initial Catalog=MyApp;Integrated Security=True"
  connection-timeout : 30s
  schema-name : akka
  table-name : EventJournal
  auto-initialize : off
  timestamp-provider : "Akka.Persistence.Sql.Common.Journal.DefaultTimestampProvider, Akka.Persistence.Sql.Common"
  metadata-table-name : Metadata
)
|INFO|Akka.Actor.RepointableActorRef|Message [LoadSnapshot] from [akka://fuse/user/MyActor#535659875] to [akka://fuse/system/akka.persistence.snapshot-store.sql-server#690509352] was not delivered. [1] dead letters encountered. If this is not an expected behavior then [akka://fuse/system/akka.persistence.snapshot-store.sql-server#690509352] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
|DEBUG|Akka.Actor.ActorCell|Stopped
|DEBUG|Akka.Actor.ActorCell|Stopped
```

**Screenshots**
![image](https://user-images.githubusercontent.com/5281560/144318322-7168ef47-e098-4678-97e9-673fb5a8dd9e.png)
![image](https://user-images.githubusercontent.com/5281560/144318389-1b8d6f15-fea9-41dd-9fb7-0268778dee0e.png)

**Environment**
Azure Cloud Service (compute emulator here), .NET Framework 4.6.1

**Additional context**
These build warnings appear in our project after upgrading, unsure if related to the issue at runtime:
```
NU1605	Detected package downgrade: System.IdentityModel.Tokens.Jwt from 6.8.0 to 5.2.4. Reference the package directly from the project to select a different version. 
 MyProj -> Akka.Persistence.SqlServer 1.4.25 -> Microsoft.Data.SqlClient 3.0.0 -> Microsoft.IdentityModel.Protocols.OpenIdConnect 6.8.0 -> System.IdentityModel.Tokens.Jwt (>= 6.8.0) 
 MyProj -> System.IdentityModel.Tokens.Jwt (>= 5.2.4)
NU1605  Detected package downgrade: System.IdentityModel.Tokens.Jwt from 5.4.0 to 5.2.4. Reference the package directly from the project to select a different version. 
 MyProj -> MyProj.Infrastructure -> DocuSign.eSign.dll 4.1.1 -> System.IdentityModel.Tokens.Jwt (>= 5.4.0) 
 MyProj -> System.IdentityModel.Tokens.Jwt (>= 5.2.4)
```

**Comment by Arkatufus** [comment]
@BobbyMorris, I managed to update 1.3.2 to 1.4.25 in the full Azure Emulator in Visual Studio with some difficulty
I'm suspecting that this is happening to you too, but the problem was that the emulator failed to copy some files to its staging folder when it tries to run the project in the emulator. These files are missing in the staging folder:
- Microsoft.Data.SqlClient.SNI.x64.dll
- Microsoft.Data.SqlClient.SNI.x64.pdb
- Microsoft.Data.SqlClient.SNI.x86.dll
- Microsoft.Data.SqlClient.SNI.x86.pdb

I managed to make it work by copying the files manually in the azure project post-build script:
```
  <PropertyGroup>
    <PostBuildEvent>
echo F|xcopy "$(SolutionDir)WorkerRole1\bin\Debug\Microsoft.Data.SqlClient.SNI.x64.dll" "$(SolutionDir)Persistent.Azure\csx\Debug\roles\WorkerRole1\approot"
echo F|xcopy "$(SolutionDir)WorkerRole1\bin\Debug\Microsoft.Data.SqlClient.SNI.x64.pdb" "$(SolutionDir)Persistent.Azure\csx\Debug\roles\WorkerRole1\approot"
echo F|xcopy "$(SolutionDir)WorkerRole1\bin\Debug\Microsoft.Data.SqlClient.SNI.x86.dll" "$(SolutionDir)Persistent.Azure\csx\Debug\roles\WorkerRole1\approot"
echo F|xcopy "$(SolutionDir)WorkerRole1\bin\Debug\Microsoft.Data.SqlClient.SNI.x86.pdb" "$(SolutionDir)Persistent.Azure\csx\Debug\roles\WorkerRole1\approot"
    </PostBuildEvent>
  </PropertyGroup>
```

It can also be caused by having both `System.Data.SqlClient` and `Microsoft.Data.SqlClient` in the same project, or a misbehaving assembly version redirect.

**Comment by Arkatufus** [comment]
This SNI deployment problem is a known issue with ADO,NET client in .NET Framework, it is officially documented in this troubleshooting guide: https://docs.microsoft.com/en-us/sql/connect/ado-net/sqlclient-troubleshooting-guide?view=sql-server-ver15#unable-to-load-native-sni-server-name-indication-library

**Comment by BobbyMorris** [comment]
@Arkatufus Thank you, I applied that and I cannot confirm that is a resolution for this because as you said we are also having issues with assembly redirects. We're working on cleaning up outdated packages that are no longer in use and updating some others, specifically those flagged as downgraded after the update to the Akka.NET projects.

**Comment by BobbyMorris** [comment]
An update, we've created demo cloud service/worker role projects to replicate this apart from our prod code base, and we can see this problem is resolved in this way. However in the prod code, even with this build step in place, it continues to happen. We're trying to see if we can duplicate this problem in the demo worker role by bringing in references to our prod code and replicating startup. We do have System.Data.SqlClient as a dependency via EF.

However, one thing we've noticed, is if the demo project has Microsoft.Data.SqlServer 4.1.0 installed, persistence will fail to come up. Is that expected? I know Akka.Persistence.SqlServer is expecting 3.0.1.

**Comment by BobbyMorris** [comment]
@Arkatufus @Aaronontheweb 
Okay, this was all Cloud Services and not Akka.Persistence.SqlServer. The other problem we had was the ref DLL from Microsoft.Data.SqlServer being copied to the deployment dir instead of the lib DLL, [as discussed here](https://github.com/Azure/azure-sdk-for-net/issues/3699#issuecomment-612306965). It just so happened we migrated from packages.config to package ref shortly before we decided to try and update Akka.NET.
